### ISSUE TITLE
Add --strict flag to treat warnings as failures

### ISSUE THREAD

**Issue by dacharyc** [agent-ecosystem/skill-validator#11]
## Problem

The current exit code behavior exits 0 when `errors == 0`, regardless of warnings. Any CI workflow that wants warnings to fail the build (e.g. for `published/` skills held to a higher standard) has to parse `-o json` output and check `.warnings > 0`. That's fragile and adds unnecessary complexity to workflow YAML.

## Proposed solution

Add a `--strict` flag to `check` and `validate structure` that exits 1 when `warnings > 0 || errors > 0`.

With this flag, the CI distinction between in-progress and published skills collapses to a single flag difference:

```sh
skill-validator check --strict published/my-skill/
skill-validator check in-progress/my-skill/
```

No JSON parsing required in either case.

## Design question

Should `--strict` be a global flag or per-command? Since `validate links` warnings (403 responses) have a different character thanShould `--stricinShould `--strict` be a global fltrShould `--strict` be a global flag or per-command? Since \mpShould `--strict`ned based on usage.
